### PR TITLE
Automatic tests

### PR DESCRIPTION
**Pull request by lilachmaliniak**
Merge automatic tests feature.

### CODE REVIEW

**Comment by lilachmaliniak** [comment]
deleted ConfigMapping.json + removed Q.fcall()

File: packager/bin/build_log [hunk 1]
```diff
@@ -0,0 +1,869 @@
+/Users/lilach.maliniak/Desktop/dev/repositories/liveDVR/packager/bin
```
Comment (kalturaguy) [inline packager/bin/build_log]: why is this file commited??

File: packager/config/nginx.conf.template [hunk 2]
```diff
@@ -49,8 +49,8 @@ http {
         vod_align_segments_to_key_frames on;
```
Comment (kalturaguy) [inline packager/config/nginx.conf.template]: big bug

**Comment by kalturaguy** [comment]
why are all binaries file commited?

**Comment by lilachmaliniak** [comment]
Because they change….by merge from 1.0.55. It might be due to changes of local build for seamless that they were different. I took to binaries from 1.0.55

File: packager/bin/build_log (hunk 1 above)
Comment (lilachmaliniak) [inline packager/bin/build_log]: removed

File: common/config/config.json.template [hunk 3]
```diff
@@ -28,7 +28,41 @@
...
+  },
+  "preserveOriginalHLS": {
+      "enable": false,
+      "createFolderPerSession": true,
+      "runDurationInMinutes": 5,
+      "validate": true,
+      "fix": true,
+      "prepareForRegression": true
+  },
+  "diagnostics": {
```
Comment (lilachmaliniak) [inline common/config/config.json.template]: removed duplicate property

File: lib/entry/FlavorDownloader.js [hunk 4]
```diff
@@ -55,7 +58,11 @@ var FlavorDownloader = (function() {
         this.downloadedChunks = {};
         this.chunksToDownload = {};
         this.loggerInformation =  "[" + that.entryId + "][" + that.flavor + "][id:" + that.id + "] ";
+        this.loggerInformation = preserveOriginalHLS ? `@@@< HLS PRESERVATION MODE >@@@ ` + this.loggerInformation : this.loggerInformation;
         this.logger =  loggerModule.getLogger("FlavorDownloader",  this.loggerInformation);
+        // sequence number used to save chunklists for automation tests
+        this.internalSequence = -1;
+        this.testAdapter = preserveOriginalHLS ? require('./../Adapters/TestAdapter.js').TestAdapter.instance : undefined;
```
Comment (kalturaguy) [inline lib/entry/FlavorDownloader.js]: please remove doesn't belong here

File: lib/Adapters/TestAdapter.js [hunk 5]
```diff
@@ -72,24 +109,26 @@ TestAdapter.prototype.getLiveEntries=function() {
...
                                 streamItem: streamItem,
                                 url: chunkListUrl
                             }
 
                     });
                     result.push({
-                        "entryId": util.format(simulateStreams.entryId,i+1),
+                        "entryId": util.format('%s-%d',path, extension),
                         "flavorParamsIds": _.keys(flavors).join(","),
                         "maxChunkCount": simulateStreams.maxChunkCount,
```
Comment (kalturaguy) [inline lib/Adapters/TestAdapter.js]: please remove maxChunkCount

File: lib/entry/FlavorDownloader.js (hunk 4 above)
Comment (lilachmaliniak) [inline lib/entry/FlavorDownloader.js]: removed

File: lib/entry/FlavorDownloader.js [hunk 6]
```diff
@@ -55,7 +58,11 @@ var FlavorDownloader = (function() {
         this.downloadedChunks = {};
         this.chunksToDownload = {};
         this.loggerInformation =  "[" + that.entryId + "][" + that.flavor + "][id:" + that.id + "] ";
+        this.loggerInformation = preserveOriginalHLS ? `@@@< HLS PRESERVATION MODE >@@@ ` + this.loggerInformation : this.loggerInformation;
```
Comment (lilachmaliniak) [inline lib/entry/FlavorDownloader.js]: removed

File: lib/Adapters/TestAdapter.js (hunk 5 above)
Comment (lilachmaliniak) [inline lib/Adapters/TestAdapter.js]: removed

File: common/config/config.json.template [hunk 7]
```diff
@@ -28,7 +28,35 @@
...
+               "history_count": 0,
+               "analyzerAlertsIgnoreList": "'INFTagDurationDoesntMatch','BitrateOutsideOfTargetAlert','InvalidKeyFrameIntervalAlert'"
+             }
+        }]
+  },
+  "preserveOriginalHLS": {
+      "enable": false,
+      "createFolderPerSession": true,
+      "runDurationInMinutes": 5,
+      "validate": true,
```
Comment (lilachmaliniak) [inline common/config/config.json.template]: removed

File: lib/entry/FlavorDownloader.js [hunk 8]
```diff
@@ -18,6 +18,9 @@ var MP4ConversionError = require('../MP4WriteStream').ConversionError;
 var ErrorUtils = require('./../utils/error-utils');
 var persistenceFormat = require('./../../common/PersistenceFormat');
 var diagnosticsAlerts = require('../Diagnostics/DiagnosticsAlerts');
+var fsUtils = require('./../utils/fs-utils');
```
Comment (kalturaguy) [inline lib/entry/FlavorDownloader.js]: fsUtils not in use?

**Pull request merged**